### PR TITLE
Upload knowledge

### PR DESCRIPTION
**Pull request by alistairknight**
# **What this PR does / why we need it**:
*A brief description of changes being made*
*Link to the relevant work items: e.g: Relates to ADO Work Item AB#213700 and builds on #3376 (link to ADO Build ID URL)*

# **Special notes for your reviewer**
*Any specific actions or notes on review?*

# Testing
*Any relevant testing information and pipeline runs.*

# Checklist (please delete before completing or setting auto-complete)
- [ ] Story Work items associated (not Tasks)
- [ ] Successful testing run(s) link provided
- [ ] Title pattern should be `{work item number}: {title}`
- [ ] Description covers all the changes in the PR
- [ ] This PR contains documentation
- [ ] This PR contains tests
- [ ] Version number in **package.json** has been updated

# **How does this PR make you feel**:
gif

### CODE REVIEW

File: app/services/upload-knowledge.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  const { buffer, type } = getBuffer(payload)
+
+  const { id } = await uploadKnowledge(buffer, type)
+
+  const metadataPayload = buildMetadataPayload(payload)
+
+  await updateKnowledgeMetadata(id, metadataPayload)
+}
+
+const saveUri = async (payload) => {
```
Comment (sfitz42) [inline app/services/upload-knowledge.js:45]: Probably should be avoiding creating placeholder functions for now. We'll come back to this once we have a plan in place.

File: app/routes/upload-knowledge.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          filename: request.payload.document.hapi.filename,
+          filetype: getExtension(request.payload.document.hapi.filename),
+          uri: request.payload.uri,
+          title: request.payload.title,
+          category: request.payload.category,
+          err
+        }).takeover(400)
+      }
+    },
+    handler: async (request, h) => {
```
Comment (sfitz42) [inline app/routes/upload-knowledge.js:42]: Need to publish a message onto the knowledge ingestion queue.

See `ADT-23`

File: app/routes/upload-knowledge.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      if (request.payload.uploadtype === 'file') {
+        await upload(request.payload)
+      } else {
+        await saveUri(request.payload)
```
Comment (sfitz42) [inline app/routes/upload-knowledge.js:46]: We will leave this for when we have a plan for URI scrapping.

**Review by sfitz42 — CHANGES_REQUESTED** [review]
Just a few changes to be addressed. This service will also need to publish a mesage on the ingestion queue in this PR.

File: app/routes/upload-knowledge.js (hunk 3 above)
Comment (alistairknight) [reply app/routes/upload-knowledge.js]: Removed for now

File: app/routes/upload-knowledge.js (hunk 2 above)
Comment (alistairknight) [reply app/routes/upload-knowledge.js:43]: Added in commit `21559f24cd8da00aabcf2dfc5ac23a3c8dad2987`

**Review by sfitz42 — COMMENTED** [review]
LGTM - Just needs a package version bump.

**Pull request merged**